### PR TITLE
improve agent performance

### PR DESCRIPTION
**Pull request by skeeey**
1. separate the controlplane binary into two binaries to reduce the binary bootstrap memory
2. strart the clusterinfo reconciles without  controller runtime and share the spoke informer with all controllers
3. use klog instead of zap logger

### CODE REVIEW

File: cmd/agent/agent.go [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
 		RunE: func(cmd *cobra.Command, args []string) error {
 			shutdownCtx, cancel := context.WithCancel(context.TODO())
 
-			level, _ := zapcore.ParseLevel(logLevel)
-			opts := &zap.Options{
-				Level: zapcore.Level(level),
-			}
-			// set log level to the controller-runtime logger
-			ctrl.SetLogger(zap.New(zap.UseFlagOptions(opts)))
+			ctrl.SetLogger(klog.NewKlogr())
```
Comment (clyang82) [inline cmd/agent/agent.go:37]: Can we have log-level in the parameter of `multicluster-agent`?

File: pkg/agent/addons/controllers/clusterclaim/clusterclaimer.go [hunk 2]
```diff
@@ -309,14 +298,8 @@ func (c *ClusterClaimer) isROSA() (bool, error) {
...
 	if err != nil {
-		if meta.IsNoMatchError(err) {
-			return false, nil
-		}
-		if strings.Contains(err.Error(), "the server could not find the requested resource") {
-			return false, nil
-		}
-		klog.Errorf("failed to mapping project:%v", err)
-		return false, err
+		klog.V(4).Infof("there is no mapping ARO cluster, %v", err)
```
Comment (clyang82) [inline pkg/agent/addons/controllers/clusterclaim/clusterclaimer.go:301]: We do not need to throw error if the environment is OpenShift or ARO. I recommend to keep it as it is.

File: cmd/agent/agent.go (hunk 1 above)
Comment (skeeey) [reply cmd/agent/agent.go:37]: we use the klog, so you can use the `--v` to set the log level

File: pkg/agent/addons/controllers/clusterclaim/clusterclaimer.go (hunk 2 above)
Comment (skeeey) [reply pkg/agent/addons/controllers/clusterclaim/clusterclaimer.go:301]: we do not need to throw error, this part just judge the type of the current platform, so I think we have a debug log is sufficient

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**